cpp/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2017 Snowflake Computing, Inc. All rights reserved.
+#
+# CMakeList for Snowflake Client++
+#
+
+set(SOURCE_FILES_CXX
+        include/snowflake++/SnowflakeInclude.hpp
```

Comment:
I believe it is better to avoid `+` in a path name. Do you have any reference?
can it be include/snowflakecpp?  or even shared with C's header file directory?

Reply:
I like include/snowflakecpp idea

Reply:
Can we put both hpp and h together in a single directory? I'm not 100% sure both C and C++ can share the same library, though, in theory, linker is supposed to pick and choose corresponding object files.